ui/src/serverIO.js

```diff
@@ -104,16 +104,12 @@ class ServerIO {
 
     this.hwrSocket.on('disconnect', (reason) => {
       console.log('hwrSocket disconnected!'); // eslint-disable-line no-console
-      const socket = this.hwrSocket;
-
-      if (reason === 'io server disconnect') {
-        setTimeout(() => {
-          socket.connect(); // try reconnecting
-        }, 500);
-      }
 
       setTimeout(() => {
-        dispatch(showConnectionLostDialog(!socket.connected));
+        dispatch(
+          // Show message if socket still hasn't reconnected (and wasn't manually disconnected in the first place)
+          showConnectionLostDialog(this.hwrSocket && !this.hwrSocket.connected),
```

Comment:
I now check that the socket is still defined. If it's not, it means the `disconnect()` method was called (and therefore that the entire app reloaded).